17600: transit_gateway: Add zone and ha_zone for azure

CyrusJavan opened the pull request:
Add a new attribute _zone_ that is used to set the availability
zone for Azure transit gateways.

Also repurpose the _ha_zone_ attribute to be used to set the
availability zone for HA Azure transit gateways.

arvindsree commented on aviatrix/resource_aviatrix_gateway.go:
```diff
@@ -938,7 +938,7 @@ func resourceAviatrixGatewayUpdate(d *schema.ResourceData, meta interface{}) err
 
-		if vpnAccess := d.Get("vpn_access").(bool); vpnAccess {
+		if vpnAccess := d.Get("vpn_access").(bool); !vpnAccess {
 			return fmt.Errorf("vpn_access must be set to yes to modify vpn authentication")
```
change to set to true

CyrusJavan committed:
17600: transit_gateway: Add zone and ha_zone for azure

Add a new attribute _zone_ that is used to set the availability
zone for Azure transit gateways.

Also repurpose the _ha_zone_ attribute to be used to set the
availability zone for HA Azure transit gateways.
